ui/selectors/transactions.js

```diff
@@ -146,7 +146,10 @@ export const transactionsSelector = createSelector(
   transactionSubSelector,
   selectedAddressTxListSelector,
   (subSelectorTxList = [], selectedAddressTxList = []) => {
-    const txsToRender = selectedAddressTxList.concat(subSelectorTxList);
+    const txsToRender = selectedAddressTxList
+      // Incoming transactions are already included in the subSelectorTxList
+      .filter((tx) => tx.type !== TransactionType.incoming)
```

Comment:
Since this selector just combines the other two, do we instead want to add this filter to the `selectedAddressTxListSelector` to avoid this issue in other places also?

Reply:
Done